internal/resolver/dns/dns_resolver_test.go

```diff
@@ -736,12 +749,122 @@ func testDNSResolver(t *testing.T) {
 	}
 }
 
+// DNS Resolver immediately starts polling on an error from grpc. This should continue until the ClientConn doesn't
+// send back an error from updating the DNS Resolver's state.
+func TestDNSResolverExponentialBackoff(t *testing.T) {
+	defer leakcheck.Check(t)
+	nt := newTimer
+	defer func(func(d time.Duration) *time.Timer) {
+		newTimer = nt
+	}(nt)
```

Comment:
There is no need for `nt`.  Or, rather, there is, but it's as the parameter to the defer, which is currently being ignored by your defer.  So, like this:

```go
	defer func(nt func(d time.Duration) *time.Timer) {
		newTimer = nt
	}(newTimer)
```

Reply:
Wow, thank you so much. My logic was just wrong lol. Fixed.